autorest/utility.go

```diff
@@ -138,11 +138,31 @@ func MapToValues(m map[string]interface{}) url.Values {
 	return v
 }
 
+// ToStringSlice method converts interface{} to []string.
+func ToStringSlice(s interface{}) []string {
+
+	v := reflect.ValueOf(s)
+	stringSlice := make([]string, v.Len())
+
+	for i := 0; i < v.Len(); i++ {
+		stringSlice[i] = v.Index(i).String()
+	}
+
+	return stringSlice
```

Comment:
[nit] the more conventional pattern would be to write the above as follows :)

``` Go
stringSlice := make([]string, 0, v.Len())

for i := 0; i < v.Len(); i++ {
    stringSlice =append(stringSlice, v.Index(i).String())
}

return stringSlice
```

Reply:
wanted to but, saw in the file its not like that so naturally assumed that's what we consider more conventional :)

Reply:
Hey, sorry for the late question but, why is this function accepts an interface, then cast elements of array/slice through `.String()`? I'll explain why I'm asking, for example, here https://github.com/Azure/azure-sdk-for-go/blob/master/services/cognitiveservices/v1.0/face/largefacelist.go#L288-L290, this function accepts several arguments one of them is `targetFace []int32`, then `autorest.Encode` called with this `[]int`, so as to result the query param `targetFace` looks like `<int32 Value> <int32 Value> <int32 Value> <int32 Value>`, because of `v.Index(i).String()`.

I've created an issue for `azure-sdk-for-go`, but also I've fixed it in my fork but didn't make a pull request. You can see that here https://github.com/thesunwave/go-autorest/commit/f36f77b4f073a6f651955e977a4ccfb01e697e38, what do you think, is it a valid solution?